### PR TITLE
Add Platform task restrictions

### PR DESCRIPTION
**Pull request by splatteredbits**

### CODE REVIEW

File: Whiskey/Whiskey.psd1 [hunk 1]
```diff
@@ -151,6 +151,8 @@
 * You can now use the [Environment class's static properties](https://docs.microsoft.com/en-us/dotnet/api/system.environment#properties) as Whiskey variables. Using these variables are now recommended over environment variables, since the existence of some environment variables varies between operating systems.
 * Added `WHISKEY_TEMP_DIRECTORY` built-in variable to get the path to the global/system temporary directory. This uses the value returned by the .NET [Path.GetTempPath()] method.
 * Added `WHISKEY_TASK_TEMP_DIRECTORY` built-in variable to get the path to the currently executing task's temporary directory, which Whiskey creates/deletes as each task runs.
+* Adding support for tasks to define what platform/operating system they can run on with a new `Platform` property on the task attribute. If a task can only run on a specific platform, set the `Platform` property to the platform is supports, e.g. `[Whiskey.TaskAttribute('SomeTask',Platform=([Whiskey.Platform]::Windows))]`. Valid platforms are `Windows`, `Linux`, and `MacOS`. The default platform is `All`.
```
Comment (DecoyJoe) [inline Whiskey/Whiskey.psd1]: Wrong tense, `Adding` should be `Added`

File: Whiskey/Whiskey.psd1 [hunk 2]
```diff
@@ -151,6 +151,8 @@
...
 * Added `WHISKEY_TEMP_DIRECTORY` built-in variable to get the path to the global/system temporary directory. This uses the value returned by the .NET [Path.GetTempPath()] method.
 * Added `WHISKEY_TASK_TEMP_DIRECTORY` built-in variable to get the path to the currently executing task's temporary directory, which Whiskey creates/deletes as each task runs.
+* Adding support for tasks to define what platform/operating system they can run on with a new `Platform` property on the task attribute. If a task can only run on a specific platform, set the `Platform` property to the platform is supports, e.g. `[Whiskey.TaskAttribute('SomeTask',Platform=([Whiskey.Platform]::Windows))]`. Valid platforms are `Windows`, `Linux`, and `MacOS`. The default platform is `All`.
+* Added `OnlyOnPlatform` and `ExceptOnPlatform` common task properties that control the platforms on which a task will/will not be run. Valid platforms are `Windows`, `Linux`, and `MacOS`.
```
Comment (DecoyJoe) [inline Whiskey/Whiskey.psd1]: The phrase "on which a task will/will not be run" looks like a typo with the double "will" slash. I think "on which a task will or will not be run" reads better.

**Comment by splatteredbits** [comment]
@DecoyJoe Ready for another review.

File: Whiskey/Whiskey.psd1 [hunk 3]
```diff
@@ -153,8 +153,9 @@
...
 * Added `WHISKEY_TASK_TEMP_DIRECTORY` built-in variable to get the path to the currently executing task's temporary directory, which Whiskey creates/deletes as each task runs.
-* Adding support for tasks to define what platform/operating system they can run on with a new `Platform` property on the task attribute. If a task can only run on a specific platform, set the `Platform` property to the platform is supports, e.g. `[Whiskey.TaskAttribute('SomeTask',Platform=([Whiskey.Platform]::Windows))]`. Valid platforms are `Windows`, `Linux`, and `MacOS`. The default platform is `All`.
-* Added `OnlyOnPlatform` and `ExceptOnPlatform` common task properties that control the platforms on which a task will/will not be run. Valid platforms are `Windows`, `Linux`, and `MacOS`.
+* Created `ConvertFrom-WhiskeyContext` and `ConvertTo-WhiskeyContext` functions for securely serializing/deserializing Whiskey's context object into background jobs. PowerShell on Linux/MacOS can't serialize `SecureString` objects. Created these functions to work around that limitation. Used by Whiskey's `Parallel` and `PowerShell` tasks.
```
Comment (DecoyJoe) [inline Whiskey/Whiskey.psd1]: Not sure why this line is showing up as being added, it should already exist on `develop`

Comment (splatteredbits) [reply Whiskey/Whiskey.psd1]: Merges. What are you going to do?

**Pull request merged**